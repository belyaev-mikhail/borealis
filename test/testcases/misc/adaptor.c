//
// Created by akhin on 6/7/16.
//

extern void  __VERIFIER_assert(int cond);
extern void  __VERIFIER_assume(int cond);
extern int   __VERIFIER_nondet_int();
extern void* __VERIFIER_nondet_pointer();
extern void  __VERIFIER_error();

int main() {
    int i = __VERIFIER_nondet_int();
    int j = __VERIFIER_nondet_int();
    char* p = __VERIFIER_nondet_pointer();

    __VERIFIER_assume(i > 0);
    __VERIFIER_assume(p);

    if (i <= 0) {
        __VERIFIER_error();
    }

    if (j <= 0) {
        __VERIFIER_error();
    }

    if (!p){
        __VERIFIER_error();
    }
}
