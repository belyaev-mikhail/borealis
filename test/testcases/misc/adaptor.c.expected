[
	{
		"location" : 
		{
			"filename" : "test/testcases/misc/adaptor.c",
			"loc" : 
			{
				"col" : 9,
				"line" : 24
			}
		},
		"type" : "UNK-99"
	}
]
